### PR TITLE
Fix pow_int()

### PR DESCRIPTION
**Pull request by pjessesco**
Exponentiation_by_squaring

### CODE REVIEW

File: src/appleseed/foundation/math/scalar.h [hunk 1]
```diff
@@ -358,7 +358,11 @@ struct PowIntHelper
 {
     static T eval(const T x)
     {
-        return x * PowIntHelper<T, P - 1>::eval(x);
+        if(P % 2 == 0)
```
Comment (dictoon) [inline src/appleseed/foundation/math/scalar.h]: Insert space between `if` and the opening parenthesis (as detailed in our [coding conventions](https://github.com/appleseedhq/appleseed/wiki/Coding-Philosophy-and-Guidelines#spacing)).

File: src/appleseed/foundation/math/scalar.h [hunk 2]
```diff
@@ -380,16 +384,25 @@ inline T pow_int(const T x)
...
 {
     // todo: implement exponentiation by squaring.
     // Reference: http://en.wikipedia.org/wiki/Exponentiation_by_squaring.
 
     T y = T(1);
 
-    while (p--)
-        y *= x;
-
+    while (p)
```
Comment (dictoon) [inline src/appleseed/foundation/math/scalar.h]: Add curly braces around multi-line body:

    while (p)
    {
        ...
    }

File: src/appleseed/foundation/math/scalar.h [hunk 3]
```diff
@@ -380,16 +384,25 @@ inline T pow_int(const T x)
...
+        if(p % 2 == 0)
```
Comment (dictoon) [inline src/appleseed/foundation/math/scalar.h]: `if (` as explained above.

File: src/appleseed/foundation/math/scalar.h [hunk 4]
```diff
@@ -380,16 +384,25 @@ inline T pow_int(const T x)
...
+        {
+            x *= x;
+            p /= 2;
+        }
+        else
+        {
+            y *= x;
+            x *= x;
+            p = (p-1) / 2;
```
Comment (dictoon) [inline src/appleseed/foundation/math/scalar.h]: Add space around all operators: `p - 1`

File: src/appleseed/foundation/math/scalar.h [hunk 5]
```diff
@@ -380,16 +384,25 @@ inline T pow_int(const T x)
 }
 
 template <typename T>
-inline T pow_int(const T x, size_t p)
+inline T pow_int(T x, size_t p)
 {
     // todo: implement exponentiation by squaring.
```
Comment (dictoon) [inline src/appleseed/foundation/math/scalar.h]: Remove todo note now that it's done (but keep the reference).

Same thing in the other function.

**Comment by pjessesco** [comment]
Ah.. no I didn't. I'll run as soon possible

**Comment by pjessesco** [comment]
All tests passed. How can I make `pow_int()` take runtime argument?

**Pull request closed without merging**